packages/nextjs/tsup.config.ts

```diff
@@ -40,9 +40,9 @@ export default defineConfig(overrideOptions => {
   const copyPackageJson = (format: 'esm' | 'cjs') => `cp ./package.${format}.json ./dist/${format}/package.json`;
 
   return runAfterLast([
-    'npm run build:declarations',
     copyPackageJson('esm'),
     copyPackageJson('cjs'),
+    'npm run build:declarations',
```

Comment:
I was seeing some weird sequencing issues where the module format-specific `package.json` files were not getting copied due to a failed declaration build, and so I shifted the ordering here to work around that.